elastic/fleet-server
# Avoid crashing on startup if Elasticsearch is not available

## What is the problem this PR solves?

Keep Fleet Server process running if Elasticsearch is not available on startup.

## How does this PR solve the problem?

* Initial `Info` request is removed, so the client can be eventually used even if during startup Elasticsearch is not available.
* Subsystems that need initialization keep retrying if they fail.
* Standalone monitor keeps running once the service is healthy. If it loses connection with Elasticsearch, it goes to degraded state.

## How to test this PR locally

* An e2e is added that uses toxiproxy to simulate connectivity loss.
* In general, this can be tested by:
  * Starting Fleet Server without connectivity with Elasticsearch (or with Elasticsearch down), it should still start, but without reaching the healthy state.
  * Start Fleet Server, and once it is healthy, interrupt connectivity with Elasticsearch.

Locally I have simulated connectivity interruptions with `iptables`.  

## Design Checklist




## Checklist





## Related issues


- Fixes https://github.com/elastic/fleet-server/issues/2683
